### ISSUE TITLE
Md-dialog without a md-dialog-actions button fail accessibility guidelines

### ISSUE THREAD

**Issue by manish2535** [angular/material#9271]
**Actual Behavior**:
 -  `What is the issue? *` - Md-dialog without a md-dialog-actions button fail accessibility guidelines. When such a dialog is opened, nothing is announced by the screen reader. The focus remains at the button which opened the dialog. If you put a dummy button in the md-dialog-actions section, then the dialog is announced and the focus management is perfect.
 -  `What is the expected behavior?` - The screen reader should announce "Enter Dialog" and put the focus on the dialog or the any other selectable element in the dialog like the close btn if no md-dialog-actions button is found. 

**CodePen** (or steps to reproduce the issue): *
- Test the "Prerendered Dialog" in the https://material.angularjs.org/latest/demo/dialog page.

**Angular Versions**: *
 -  `Angular Version:` - v1.5.5
 -  `Angular Material Version:` - v1.1.0-rc.5

**Additional Information**:
 -  `Browser Type: *` - Chrome 
 -  `Browser Version: *` - Version 51.0.2704.103 m
 -  `OS: *` - Windows 7
 -  `Stack Traces:`
